ekundayo-ab/hello-books
# #149786054 Assign Membership Levels

### What does this PR do?
It ensures a user's membership level is automatically upgraded if certain conditions are met

### Description of Task to be completed?
1. Membership levels with details as below should be incorporated:
- **Bronze level** 2 credits, upgraded to **Silver** after borrowing **10** times
- **Silver level** 3 credits, upgraded to **Gold** after borrowing **20** times
- **Unlimited level** 9005 credits no upgrade for now
2. For every borrowing a user makes, 1 credit should be deducted from user's level credit
3. After the book is returned, the 1 credit should be redeemed back.
4.  User should not be able to borrow if credit point is 0

### How should this be manually tested?
1. Pull in latest changes from remote repository to local repository
2. Fire up terminal run `npm install`
3. execute `npm run start:dev` and then `npm run build:dev` to start up application
4. Visit url supplied by the application usually **http://localhost:3000**
5. login with `ekprogs@gmail.com` and `dayo` for admin access, while for normal user access simply register a new user on the application
6. After logging in to the application, borrow and return books, subsequently check your profile to observe the borrowing count.
7. After count reaches 10 and a request is made in the application, an auto upgrade notification should be seen with effected level visible on user's profile

### Any background context you want to provide?
The credit points are made as it is for demo purposes, no upgrade support yet for an unlimited user

### What are the relevant pivotal tracker stories?
#149786054 User should be assigned membership levels

### Screenshots (if appropriate)
N/A

### Questions:
N/A
